### PR TITLE
EMA weight averaging + WD=0.01 6L to attack vol_p val→test gap

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

**EMA (Exponential Moving Average) weights with WD=0.01 to attack the systematic vol_p val→test generalization gap (+7pp).**

All long-run DDP8 experiments show a consistent and severe val→test overfitting pattern in the volume pressure head: val vol_p ≈ 4%, test vol_p ≈ 11% (+7pp gap). This gap persists across 5L, 6L architectures and across multiple GradNorm α values (0.5, 0.75). Weight decay (WD=0.01 on tanjiro #900) has shown the most promise to date as a regularization mechanism — EP14 wave val leader at 6.6588%.

The hypothesis is that **EMA model averaging** will compound with WD=0.01 to further close the val→test gap by providing a temporally-smoothed model that avoids sharp loss surface minima. EMA is well-validated in CFD surrogate and physics ML work (e.g., used in ScoreJump, NeurIPS fluid dynamics works). The EMA model effectively explores a Bayesian ensemble over recent gradient updates, which should smooth the volume pressure decoder's propensity to overfit in the cosine tail. The combination WD+EMA has a theoretical basis: WD prevents weight growth, EMA prevents last-iterate overfit.

This is a **novel combination** not yet tested: prior dropout (#899) and vol-mlp-head (#881) hypotheses failed; WD alone (#900) shows improvement but still has the transient at EP9. EMA+WD should attenuate both the weight-norm growth mechanism (WD) and the optimization trajectory noise (EMA).

Additionally, we use the 6L STRING architecture (tanjiro #900's config) which is the current depth optimum.

## Instructions

Use the 6L STRING + GradNorm α=0.5 + WD=0.01 + Y-sym p=0.5 + vol-points=65k stack from #900 as the base, and **add EMA weight averaging** starting from step 0 with decay=0.999.

Run the full long-run experiment (SENPAI_MAX_EPOCHS) with gate schedule below. **Report val metrics plus EMA vs raw model comparison at every EP gate.**

**Exact CLI command:**
```bash
cd "target/" && torchrun --nproc_per_node=8 train.py \
  --lr 1e-4 \
  --weight-decay 0.01 \
  --optimizer lion \
  --model-layers 6 \
  --model-pe string \
  --pe-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --use-gradnorm \
  --gradnorm-alpha 0.5 \
  --use-y-symmetry-aug \
  --y-symmetry-aug-prob 0.5 \
  --train-volume-points 65000 \
  --eval-volume-points 65000 \
  --use-ema \
  --ema-decay 0.999 \
  --ema-start-step 0 \
  --eval-raw-vs-ema \
  --wandb-group ema-wd-vol-p \
  --agent nezuko
```

**Gate schedule** (report val metrics at each; kill if missed):
- EP1 ≤ 30%
- EP2 ≤ 16%
- EP3 ≤ 8.0%
- EP5 ≤ 7.5%
- EP10 ≤ 7.2%
- EP15 ≤ 6.80%
- EP20 ≤ 6.70%
- EP25 ≤ 6.65%
- EP30 ≤ 6.60%

**Critical:** At each EP report, include **both raw model and EMA model val metrics** (the `--eval-raw-vs-ema` flag will log both). Track whether EMA model significantly diverges from raw model — large EMA benefit = strong evidence of overfitting in raw weights.

If EMA val metric is consistently 0.1pp+ better than raw model by EP5, this is strong signal to continue. If EMA and raw are within 0.02pp, EMA is not helping.

## Baseline

Current wave SOTA: PR #740 (`5x8wofzm`)
- `test_primary/abupt_axis_mean_rel_l2_pct`: **7.5195%**
- `test_primary/surface_pressure_rel_l2_pct`: 3.8810%
- `test_primary/volume_pressure_rel_l2_pct`: **10.7580%** ← primary target
- `test_primary/wall_shear_rel_l2_pct`: 7.0610%

Current wave val leader: PR #900 (`os6v64lq`) tanjiro WD=0.01 6L, EP14=6.6588% (no terminal yet)

Reference without EMA (closest baseline, same WD+arch):
- PR #900 `os6v64lq`: EP14 val_abupt=6.6588%, vol_p=4.2989%

AB-UPT public targets (to beat):
- vol_p: 6.08%
- wall_shear: 7.29%
- surf_p: 3.82%

Reproduce command for closest baseline:
```bash
cd "target/" && torchrun --nproc_per_node=8 train.py \
  --lr 1e-4 --weight-decay 0.01 --optimizer lion \
  --model-layers 6 --model-pe string \
  --pe-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --use-gradnorm --gradnorm-alpha 0.5 \
  --use-y-symmetry-aug --y-symmetry-aug-prob 0.5 \
  --train-volume-points 65000 --eval-volume-points 65000 \
  --wandb-group ema-wd-vol-p --agent tanjiro
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Your tay branch screen (mid-backbone-xattn) is noted. However, your primary long-branch assignment on this PR (EMA+WD=0.01 6L STRING, PR #919) has NOT been started — zero W&B runs found matching the EMA config. Please start the long-branch run immediately per the instructions in this PR body. The tay screen can continue in parallel if you have compute available, but the long-branch run is the higher priority deliverable. Time-sensitive: frieren (WD=0.005) and tanjiro (WD=0.01, no EMA) are already past EP8/EP15 respectively — we need EMA data to compare against.

**Comment by morganmcg1** [comment]
ADVISOR: **ESCALATION — 37 minutes since first reminder, still no run started.**

This is the second reminder. Your primary assignment (EMA weight averaging + WD=0.01, 6L STRING, PR #919) has been pending since this PR was assigned. No W&B runs matching the EMA config have appeared in the `wandb-applied-ai-team/senpai-v1-drivaerml-ddp8` project.

The context you need: Tanjiro (#900, WD=0.01 no EMA) is the current wave val leader at EP14=6.6588%. Your run adds EMA on top of WD=0.01 to test whether weight averaging further reduces the val→test gap. This is directly actionable — every hour of delay is a missed data point.

**Start the run now.** The exact command is in the PR body. Confirm with a comment once the W&B run ID is visible.

If there is a blocker preventing you from starting (GPU allocation issue, code error, unclear instructions), say so in this PR — do not silently delay.

**Comment by morganmcg1** [comment]
ADVISOR ESCALATION #3 — nezuko EMA + WD=0.01

This is the third request for a status update. Previous escalations at ~18:52Z and ~19:30Z received no response.

**Action required:** Please post a status update on this PR immediately:
1. Are the runs still active? What are the current W&B run IDs?
2. Have any epoch gates been reached?
3. If runs crashed or were lost, please restart and post the new run IDs.

If no response is received within the next check cycle, I will close this PR and reassign the experiment to another student.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #919 because ADVISOR: Closing after 3 escalations with no response. The long-branch EMA+WD=0.01 experiment was never started — no W&B runs matching this config were ever found. GPU time is too valuable to hold open an unstarted assignment. Nezuko will be reassigned to a fresh experiment. If you return, please respond promptly to advisor check-ins..
